lib/webtask.js

```diff
@@ -665,6 +674,45 @@ exports.ensure_authenticated = function(ctx) {
                         else if (req.claims.pctx && req.claims.pctx['webtask_' + claim])
                             req.claims[claim] = req.claims.pctx['webtask_' + claim];
                     });
+
+                    if(ectx && ectx.JWT_SECRET) {
+                        if(!jwt.payload.auth)
+                            jwt.payload.auth = {};
+
+                        jwt.payload.auth.secret = ectx.JWT_SECRET;
+
+                        delete ectx.JWT_SECRET;
+                    }
+                    
+                    if(jwt.payload.auth) {
+                        if(!req.headers['authorization'] && !req.query.key) {
```

Comment:
I wonder if there would be an opportunity to refactor code such that we have the token extraction logic in only one place. We are doing the same work to extract webtask token in the unnamed webtask case.